### PR TITLE
Adding packages, UserController & connected classes

### PR DESCRIPTION
**Pull request by karolpolikarp**
Adding packages for domain, controller, repository and service.
Adding classes User, UserDto, UserController, UserRepository, UserService and UserNotFoundException.

Method generateUserIdKey has to be expanded in next commit to make IdKey valid for an hour after successful login.

### CODE REVIEW

File: src/main/java/com/kodilla/ecommercee/controller/UserController.java [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+
+    @Autowired
+    private UserMapper userMapper;
+
+    @Autowired
+    private UserService userService;
+
+    @RequestMapping(method = RequestMethod.GET, value = "getUsers")
+    public List<UserDto> getUsers() {
+//        return userMapper.mapToUserDtoList(userService.getUsersList());
```
Comment (interceptor2048) [inline src/main/java/com/kodilla/ecommercee/controller/UserController.java]: Zakomentowane linie nie mogą wejść do mastera.

File: src/main/java/com/kodilla/ecommercee/repository/UserRepository.java [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+
+import java.util.List;
+import java.util.Optional;
+
+public interface UserRepository extends CrudRepository<User, Long> {
+
+    @Override
+    List<User> findAll();
+
+    Optional<User> findById(Long id);
```
Comment (interceptor2048) [inline src/main/java/com/kodilla/ecommercee/repository/UserRepository.java:15]: Użyj metody z CrudRepository i daj @Override.

**Review by interceptor2048 — CHANGES_REQUESTED** [review]

File: src/main/java/com/kodilla/ecommercee/repository/UserRepository.java (hunk 2 above)
Comment (karolpolikarp) [reply src/main/java/com/kodilla/ecommercee/repository/UserRepository.java:15]: Dodałem @Override do Optional<T> findById(ID id)

File: src/main/java/com/kodilla/ecommercee/controller/UserController.java (hunk 1 above)
Comment (karolpolikarp) [reply src/main/java/com/kodilla/ecommercee/controller/UserController.java]: Zakomentowane linie zostały usunięte.

**Review by interceptor2048 — APPROVED** [review]

**Pull request merged**